pkg/riff/commands/namespace.go

```diff
@@ -153,3 +154,53 @@ func NamespaceCleanup(c *core.Client) *cobra.Command {
 
 	return command
 }
+
+func NamespaceAddCredentials(c *core.Client) *cobra.Command {
+	options := core.NamespaceAddCredentialsOptions{}
+
+	command := &cobra.Command{
+		Use:     "add-credentials",
+		Short:   "add secret and bind it to the " + env.Cli.Name + " service account",
+		Example: `  ` + env.Cli.Name + ` namespace add-credentials --secret build-secret --namespace default --docker-hub johndoe`,
+		Args:    cobra.ExactArgs(namespaceAddCredentialsNumberOfArgs),
+		PreRunE: FlagsValidatorAsCobraRunE(
+			FlagsValidationConjunction(
+				ValidDnsSubdomain("namespace"),
+				AtMostOneOf("gcr", "docker-hub", "registry-user"),
+				FlagsDependency(Set("registry-user"), NotBlank("registry")),
+				FlagsDependency(Set("registry"),
+					NotBlank("registry-user"),
+					SupportedRegistryProtocol(func() string {
+						return options.Registry
+					})),
+				NotBlank("secret"),
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := (*c).NamespaceAddCredentials(options); err != nil {
+				return err
+			}
+			PrintSuccessfulCompletion(cmd)
+			return nil
+		},
+	}
+
+	command.Flags().StringVar(&options.NamespaceName, "namespace", "", "the `namespace` of the credentials to be added")
+	command.Flags().StringVarP(&options.SecretName, "secret", "s", "", "the name of a `secret` containing credentials for the image registry")
```

Comment:
What if we default the secret to `push-credentials` to update the credentials when the namespace is first inited?

Reply:
As mentioned above, this command adds secrets, but does not update them.

Reply:
Now the command allows secret update. While I see the proposed symmetry with `namespace init`, it seems a bit strange to me to allow to write `credentials set` without naming them at all. Happy to revisit if you think it's more consistent that way.